content-keys-test-vectors.md

```diff
@@ -0,0 +1,142 @@
+state_root = 0xd1c390624d3bd4e409a61a858e5dcc5517729a9170d014a6c96530d64dd8621d 
+```
+
+##### Expected Output
+```
+content_key = 0x02829bd824b016326a401d083b33d092293333a830d1c390624d3bd4e409a61a858e5dcc5517729a9170d014a6c96530d64dd8621d
+content_id = 0x6427c4c8d42db15c2aca8dfc7dff7ce2c8c835441b566424fa3377dd031cc60d
+content_id: U256 = 45301550050471302973396879294932122279426162994178563319590607565171451545101
+```
+
+#### ContractStorageTrieProof
+
+##### Input Parameters
+```
+address = 0x829bd824b016326a401d083b33d092293333a830
+slot = U256(239304)
+state_root = 0xd1c390624d3bd4e409a61a858e5dcc5517729a9170d014a6c96530d64dd8621d
+```
+
+##### Expected Output
+```
+content_key = 0x03829bd824b016326a401d083b33d092293333a830c8a6030000000000000000000000000000000000000000000000000000000000d1c390624d3bd4e409a61a858e5dcc5517729a9170d014a6c96530d64dd8621d
+content_id = b1c89984803cebd325303ba035f9c4ca0d0d91b2cbfef84d455e7a847ade1f08
+content_id: U256 = 80413803151602881485894828440259195604313253842905231566803078625935967002376
+```
+
+#### ContractBytecode
+
+##### Input Parameters
+```
+address = 0x829bd824b016326a401d083b33d092293333a830
+code_hash = 0xd1c390624d3bd4e409a61a858e5dcc5517729a9170d014a6c96530d64dd8621d
+```
+
+##### Expected Output
+```
+content_key = 0x04829bd824b016326a401d083b33d092293333a830d1c390624d3bd4e409a61a858e5dcc5517729a9170d014a6c96530d64dd8621d
+content_id = 0x146fb937afe42bcf11d25ad57d67734b9a7138677d59eeec3f402908f54dafb0
+content_id: U256 = 9243655320250466575533858917172702581481192615849913473767356296630272634800
+```
+
+### History Network Keys
+
+#### HeaderKey
+
+##### Input Parameters
+```
+chain_id = 15
+block_hash = 0xd1c390624d3bd4e409a61a858e5dcc5517729a9170d014a6c96530d64dd8621d
+```
+
+##### Expected Output
+```
+content_key = 0x000f00d1c390624d3bd4e409a61a858e5dcc5517729a9170d014a6c96530d64dd8621d
+content_id = 0x2137f185b713a60dd1190e650d01227b4f94ecddc9c95478e2c591c40557da99
+content_id: U256 = 15025167517633317571792618561170587584740338038067807801482118109695980329625
+```
+
+#### BodyKey
+
+##### Input Parameters
+```
+chain_id = 20
+block_hash = 0xd1c390624d3bd4e409a61a858e5dcc5517729a9170d014a6c96530d64dd8621d
+```
+
+##### Expected Output
+```
+content_key = 0x011400d1c390624d3bd4e409a61a858e5dcc5517729a9170d014a6c96530d64dd8621d
+content_id = 0x1c6046475f0772132774ab549173ca8487bea031ce539cad8e990c08df5802ca
+content_id: U256 = 12834862124958403129911294156243112356210437741210740000860318140844473844426
+```
+
+#### ReceiptsKey
+
+##### Input Parameters
+```
+chain_id = 4
+block_hash = 0xd1c390624d3bd4e409a61a858e5dcc5517729a9170d014a6c96530d64dd8621d
+```
+
+##### Expected Output
+```
+content_key = 0x020400d1c390624d3bd4e409a61a858e5dcc5517729a9170d014a6c96530d64dd8621d
+content_id = 0xaa39e1423e92f5a667ace5b79c2c98adbfd79c055d891d0b9c49c40f816563b2
+content_id: U256 = 76995449220721979583200368506411933662679656077191192504502358532083948020658
```

Comment:
What is the reason for presenting this as both an integer and the hex value?

This de-normalization seems a little un-necessary and has the potential downside of the two values not actually being the same due to clerical errors.

Reply:
The reason is because U256 is the actual type of the content-id so it'd be nice to have it there but not all languages support it natively, like JS, so we have the hex value as well.